doc/sphinx/source/community.rst

```diff
@@ -0,0 +1,175 @@
+          through it is still under development.
+
+Whitespaces and indentation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+#. Do not use tabs. Only spaces are allowed.
+#. Set indent to 4 spaces per indent.
+#. Indent ``public``, ``private``, ``protected`` with the same indent as
+   a ``class`` block.
+#. Indent ``class`` block with the same indent as a namespace:
+
+   ::
+
+     // OK
+     namespace foo
+     {
+
+     class entity
+     {
+     public:
+         /* */
+     private:
+         /* */
+     protected:
+         /* */
+     };
+
+     } // namespace foo
+
+#. Indent switch blocks so that the ``case X:`` statements are indented in the switch block:
+
+   ::
+
+     // OK
+     switch (foo) {
+         case 1:
+             a += 1;
+             break;
+
+         case 2: {
+             a += 2;
+             break;
+         }
+     }
+
+#. Do not indent labels.
+#. Do not indent preprocessor block.
+#. Insert space around operators.
+#. Insert space padding between ``if``, ``for``, ``while`` and the following bracket.
+#. **Do not leave trailing whitespaces.**
+
+Bracket placement
+~~~~~~~~~~~~~~~~~
+
+#. Use `One True Bracket Style`_.
+#. Place brackets in namespaces, classes, extern-"C" and functions on the next line, e.g:
+
+   ::
+
+     // Wrong
+     namespace foo {
+
+     }
+
+     // OK
+     namespace bar
+     {
+
+     }
+
+#. Add brackets to unbracketed one line conditional statements (e.g. ``if``, ``for``, ``while``...).
+#. Close whitespace in the angle brackets of template definitions:
+
+   ::
+
+     foo<int, bar<int>> obj;
+
+Code length
+~~~~~~~~~~~
+
+#. Prefer 80 character code line length. 100 characters is a maximum allowed line length.
+#. Prefer to keep functions and methods with not more than 100 lines of code.
+
+Naming
+~~~~~~
+
+#. Use `snake_case`_ for identifer names.
+#. Use ``m_`` prefix when naming private and protected class members.
+#. Use CamelCase style only for template parameters:
+
+   ::
+
+     template<typename CrazyAllocator>
+     class my_strange_container;
+
+Comments
+~~~~~~~~
+
+#. **Do not leave commented code.**
+#. Prefer not keeping ``TODO`` in code without corresponding issue.
+   Place issue number next to the ``TODO`` with a little explanation why it is there:
+
+   ::
+
+     // TODO #123: replace foo() with something that exposes smaller IRQ latency.
+     foo();
+
+#. Use doxygen for commenting the function prototypes and class methods.
+#. Use C++ comment lines for doxygen commenting:
+
+   ::
+
+     //! This function does something.
+     //! \details Also it calls 911.
+     void foo();
+
+#. Do not use ``\brief`` doxygen directive. Brief tag is set automatically for first comment line.
+
+Commit and PR style
+-------------------
+
+.. highlight:: none
+
+#. Try to follow common `git guidelines`_.
+#. **Do not keep commits named just** ``FIX`` **,** ``Review`` **,** ``tmp`` **, etc.**
+#. Prefer placing issue number at the start of the summary line, prefixed with ``>#``:
```

Comment:
A line with the `#` at the start is interpreted as the commentary. Even first line. And `>#` is even coolish.